tools/object-storage/upload_download.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2024 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/go-kit/log"
+	"github.com/thanos-io/objstore"
+	"github.com/thanos-io/objstore/client"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+type Store struct {
+	TsdbPath     string
+	ObjectKey    string
+	ObjectConfig string
+}
+
+func newstore() *Store {
```

Comment:
This constructor should consume the parameters from the command line and initialize the bucket here.

Copying from your code.
```go
    config := c.ObjectConfig
	configBytes, err := os.ReadFile(config)
	if err != nil {
		panic(fmt.Errorf("failed to read config file: %v", err))
	}
	commitDir := c.ObjectKey
	bucket, err := client.NewBucket(log.NewNopLogger(), configBytes, "example")
	if err != nil {
		panic(err)
	}
```

And you can feed the args from CLI.